### PR TITLE
adding unit tests to TestConfiguration file

### PR DESCRIPTION
**Pull request by katarinarosiak**
Co-authored-by: Tim Dronkers <tdronkers@gmail.com>
Co-authored-by: Scott Graham <scttgrhm7+public@gmail.com>
Co-authored-by: Miles Abbason <miles.abbason@gmail.com>

This PR: 
- reactors `checkJsonBody` function and helpers to handle edge cases such as when `property` is `null` or `undefined`. 
- adds `__tests__` folder in the root directory with a test file and two JSON files: `test.json` and `response.json`. Those files help to test how the instance of `TestConfiguration` will work with the example test and response.     
- adds a test suit for test `TestConfiguration` file, especially `checkJsonBody`, `checkHeaders` and `checkResTimeOrStatusCode` functions.
 

You can now run tests for this file with the command `npm test`. 
The current coverage is:
Screenshot-from-2022-08-01-15-19-29

(I'm sure there are several edge cases that I haven't thought about yet, so if anything pops up along the way, just let me know.)

The Test Runner has been tested locally with the result: 
```javascript
SHAPE OF RESULTS ------> [
  {
    assertionId: 194,
    assertionType: 'statusCode',
    targetValue: '200',
    actualValue: 200,
    comparisonType: 'equalTo',
    property: null,
    success: true
  },
  {
    assertionId: 195,
    assertionType: 'body',
    targetValue: 'end2end-prototype',
    actualValue: 'end2end-prototype',
    comparisonType: 'equalTo',
    property: '$[0].title',
    success: true
  },
  {
    assertionId: 196,
    assertionType: 'header',
    targetValue: 'close',
    actualValue: 'close',
    comparisonType: 'equalTo',
    property: 'connection',
    success: true
  },
  {
    assertionId: 197,
    assertionType: 'responseTime',
    targetValue: '100',
    actualValue: 726,
    comparisonType: 'lessThan',
    property: null,
    success: false
  }
]
```

### CODE REVIEW

File: coverage/clover.xml [hunk 1]
```diff
@@ -0,0 +1,220 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (scogra17) [inline coverage/clover.xml:1]: Are these supposed to be included in the repo or can they be added to the `.gitignore`?

**Review by scogra17 — COMMENTED** [review]
Looks great @katarinarosiak! I'm wondering, can we add `/coverage` to `.gitignore`?

**Review by mabbason — APPROVED** [review]
Nice Katarina! Good to have these tests 👍
